modules/createHashHistory.js

```diff
@@ -83,7 +87,7 @@ const createHashHistory = (options = {}) => {
   }
 
   const createHref = (path) =>
-    '#' + history.createHref(path)
+    '#' + transformPath(history.createHref(path), true)
```

Comment:
I understand your reasoning, but after having tried to implement this I came to the realisation that this would mean it in fact wouldn't be a hash history at all anymore. It would mean users would have to specify their own version of the [getHashPath()](https://github.com/tvervest/history/blob/escaped-fragment-support/modules/HashProtocol.js#L16) method. I suggest we enforce a `#` in the URL, but everything after that is customisable by the user's transformation method.